### ISSUE TITLE
NoMethodError: undefined method `describe'

### ISSUE THREAD

**Issue by logicminds** [metaskills/minitest-spec-rails#72]
```
bundle install
...
Using sass-rails 5.0.4
Using rails 4.2.6
Using web-console 2.3.0
Using jquery-datatables-rails 3.4.0
Using minitest-spec-rails 5.3.0

rake test test/controllers/gig_confirmation_controller_test.rb
NoMethodError: undefined method `describe'
```

```
require 'test_helper'
class GigConfirmationsControllerTest < ActionController::TestCase
describe :create do
    let(:data) do
      {
        gig_confirmation: {
          gig_id: gig.id,
          confirmer_email: 'a@a.com',
          confirmer_first_name: 'Some Text',
          confirmer_last_name: 'Some Text',
          confirmer_role: 'Role',
          message: false
        }
      }
    end
    let(:gig) do
      @gig1_post1_profile1_admin
    end

    describe 'not logged in' do
      it 'should redirect to login' do
        post :create, data
        assert_redirected_to :login
        assert_equal 'Please login.', flash[:danger]
      end
    end
end
end
```

**Comment by metaskills** [comment]
Thanks for the report. Everything above looks fine. This I have never passed a symbol to describe before. FWIW, in many apps and in our tests, Rails 4.2 and upward works fine. May I suggest debugging this a bit with the following case. Does this work?

```ruby
require 'test_helper'
class GigConfirmationsControllerTest < ActionController::TestCase
  it 'test' do
    assert true
  end
end
```

**Comment by logicminds** [comment]
The code above gets me the following error:

```
NoMethodError: undefined method `it' for GigConfirmationsControllerTest:Class
/myapp/test/controllers/sample_controller_test.rb:3:in `<class:GigConfirmationsControllerTest>'
/myapp/test/controllers/sample_controller_test.rb:2:in `<top (required)>'
/home/developer/vendor/ruby/2.2.0/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/home/developer/vendor/ruby/2.2.0/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `block in require'
/home/developer/vendor/ruby/2.2.0/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:240:in `load_dependency'
/home/developer/vendor/ruby/2.2.0/gems/activesupport-4.2.6/lib/active_support/dependencies.rb:274:in `require'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:114:in `block (3 levels) in define'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:114:in `each'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:114:in `block (2 levels) in define'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:113:in `each'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:113:in `block in define'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:31:in `create_and_run_single_test'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/sub_test_task.rb:17:in `invoke_rake_task'
/home/developer/vendor/ruby/2.2.0/gems/railties-4.2.6/lib/rails/test_unit/testing.rake:8:in `block in <top (required)>'
```

**Comment by logicminds** [comment]
There is nothing to require?  Just install and go right?

**Comment by logicminds** [comment]
Does this work on controllers?  Examples specify ActiveSupport::TestCase

**Comment by metaskills** [comment]
> There is nothing to require? Just install and go right?

Correct, the Railtie loads and does all the work for you.

 > Does this work on controllers? Examples specify ActiveSupport::TestCase

Oh yea. All those other things like `ActionController::TestCase` inherit from `ActiveSupport::TestCase` and the behavior for minitest-spec-rails is mixed in there. Does a unit/model test work?

**Comment by logicminds** [comment]
Ok so it seems to be something with my rails environment.


running `RAILS_ENV=test rake test test/controllers/gig_confirmation_controller_test.rb` works fine.  As does the model test.

However, when switching to RAILS_ENV=development I get the errors above.

**Comment by metaskills** [comment]
Running `./bin/rake test ...` should not be done with a RAILS_ENV. Is there a reason you are doing that. Rails needs to do things like clone your dev env and change things for the test process for you. That's somewhat changed in Rails 5. But curious if just doing `./bin/rake test filename.rb` works for you?

**Comment by logicminds** [comment]
I am using docker containers so I have ephemeral web and database containers. 

As long as you have the proper databases already created switching rails environments should not be a problem. AFAIK.

Even without containers this is still a problem on my local environment

**Comment by metaskills** [comment]
Interesting. I just did a few more tests on my own stuff. If I do...

```
$ RAILS_ENV=development ./bin/rake test test/models/foo.rb
```

I can see the issue you described. Part of me believes that development should be implied and that Rails should default to that and be allowed to control that when it spawns the test env. The other part of me might think that removing the `Rails.env.test?` stuff here would be good.

https://github.com/metaskills/minitest-spec-rails/blob/master/lib/minitest-spec-rails/railtie.rb

Can you confirm that would help for you? Maybe I can do a PR and see how Travis tests hold up.

**Comment by logicminds** [comment]
Ok well yea thats pretty explicit and would explain why none of this worked.  I am somewhat new to rails development but not ruby. 

Is it not standard practice to run tests in development to validate my crap code without switching to the test env?

**Comment by metaskills** [comment]
> Is it not standard practice to run tests in development to validate my crap code without switching to the test env?

It is... but I've never seen `RAILS_ENV` set to development before. Just always assumed it was best to let Rails set the default and then switch using the top of `test_helper.rb`. Also, I am almost certain you don't have crap code :)

So... I clearly think this is a bug in what we are doing. I made PR #73 and made a mention we put these guards in for "some" reason back in v4.3.6 of the gem. If I remember correctly, the reason we did that was it was confusing for some people to see blank test runner output when they bumped up a Rails console (development mode) and exited. Which would happen because these Railties would kick in.

What normally happens in vanilla Rails is the `test_helper.rb` requires `rails/test_help.rb` which does all the requires for ActiveSupport test case and hence all the Minitest autorun stuff as well, which is the output you would see in the things I mentioned above. I'm not sure what the best way to solve this is. Do you have any thoughts?

**Comment by metaskills** [comment]
So I see a few options:

* Do not explicitly set RAILS_ENV=development on your end. If this is the solution I will update the readme.
* Explore if our `Rails.env.test?` guards to keep railtie behavior out of development mode can be done in other ways. For example, maybe just using `ENV['RAILS_ENV']` is better since the `Rails.env` string inquirer is memorizes. This feels worth exploring.
* Make users put a require at the top of `test_helper.rb` under the `rails/test_help` require. Not feeling this, but wanted to put it out there.

**Comment by metaskills** [comment]
OK... I just published v5.4.0 which uses `ENV['RAILS_ENV']`. This does not fix things 100% since the way Bundler/Rails works is that you likely have (properly) mintiest-spec-rails within the `test` group. This means it would not be required for you when you manually set `RAILS_ENV` to development before running tests. I would suggest not doing that and allow Rails to use the default env as needed. Even setting the env to `test` is an anti-pattern IMO to since Rails wants to do this for you so it can do all the crazy work of cloning DB structure, etc. That said, I have done it when needed but only when I we are seriously monkey patching the database tasks in ActiveRecord for big legacy DB concerns : / 

That said, if you do have a need to set these ENVs specifically, the v.5.4.0 will help you with that. All you need to do is make your test helper look like this. Note the require for our gem. Closing this issue, but feedback still very much welcome. Cheers!!! 

```ruby
ENV["RAILS_ENV"] = "test"
require File.expand_path('../../config/environment', __FILE__)
require 'rails/test_help'
require 'minitest-spec-rails'

class ActiveSupport::TestCase
```

**Comment by logicminds** [comment]
Thanks, you are awesome!
